### ISSUE TITLE
Homebrew formula for daemon

### ISSUE THREAD

**Issue by gabriel** [keybase/go#215]
I'm creating a home brew formula for the daemon (keybased) ... The OSX UI will give the option to install when it runs (if it isn't already installed). It will also check for some minimum reported version/build.

**Comment by malgorithms** [comment]
nice! 

As for "option to install",  is presenting this as an option temporary and/or for dev purposes?

I ask because I would like, in the long run, for keybase to come with everything it needs or download without asking. I don't think a user should ever have to see the term `keybased` or think about what service components are installed. For related examples, a user doesn't have to worry about `ChromeHelper` or `Dropbox Finder Integration`

In the long run, what we should aim for is one install of Keybase that gets you everything.
 - the GUI
 - the CLI, properly in your path
 - kbfs + keybased, both set to start up automatically
 
So, ideally, someone has keybase or not. Not parts of it, or a mishmash of versions they might piece together. Obviously our requirements as devs may be different.

Similarly, the "brew install keybase" should get someone everything - you wouldn't run one brew install to get the daemon, one to get the CLI, one to get kbfs, etc.

**Comment by gabriel** [comment]
Yeah ideally we will run the install automatically from the GUI without prompting... I'm thinking in debug versions I'm gonna have it prompt just to have it in our faces.

So this has to work even if people don't have homebrew installed so I'm thinking we can either:
(a) If homebrew isn't installed, install it and then run brew keybase
(b) Don't install homebrew but install keybase the same way as homebrew (so it's compatible)

Also agree on having a single brew formula for installing daemon/client etc.

**Comment by maxtaco** [comment]
Gabriel, are you saying that when we ship, we're going to depend upon Homebrew?  I don't know much about OSX install, but that feels fragile to me.

**Comment by malgorithms** [comment]
in release versions, I think installing brew for someone would be problematic. Just from my own playing with brew on different machines, every now and then something is fucked and I have to mess around with `brew doctor` to get it fixed. Having this happen automatically is a bit scary...and I think keybase for regular users needs to never require a terminal.

Installing keybase where brew does -- I have no idea what the concerns there would be. Might brew mess with it? Overwrite it? Have errors? What happens if you uninstall brew?

what if:

1. the regular version installs somewhere outside brew, like most apps...but if you try to install it and the brew version is installed, it gives you an error.

2. our official brew release will look for our standard install location and do the opposite?  i.e., brew won't let you install keybase if it's already installed outside brew

**Comment by gabriel** [comment]
ok yeah this makes sense... we don't want dependencies on homebrew.. so:

- Check to see if keybase is already installed via homebrew. If so, use that.
- If not installed, install it (in a standard osx way that doesn't require homebrew).

If somehow you get 2 versions installed, 1 via brew and 1 via ui installer, then we'll show a warning or error. This shouldn't happen though but we should protect for it.

So I'm gonna do the UI installer and hold off on brew support since it's not required yet.
